specs/state-proof.md

```diff
@@ -0,0 +1,351 @@
+# State Proof
+
+In EVM the interpreter has ability to do any random read-write access to data like account balance, account storage, or stack and memory in current scope, but it's hard for a circuit to keep tracking these data to ensure their consistency from time to time. So the state proof helps EVM proof to check all the random read-write access records are valid, throgh grouping them by their unique index first, and then sorting them by accessing timestamp. We call the accessing timestamp **global counter**, which counts the number of access records and also serves as an unique identifier for a record.
```

Comment:
The timestamp sounds like date-time related. What do you think we rephrase "accessing timestamp" to "order of access"?

Reply:
Should we also define `global counter` as a variable somewhere and give it an `int` type?

Reply:
"Order of Access" sounds great!

Do you think we should have a file for all terminology or we separate the terminology in the file who defines them?

Reply:
the latter sounds better.